### PR TITLE
Include the kroxylicious maintained filters in the dist by default

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

Currently, we activate a profile to include the kroxylicious maintained filters in the distributions.  Following discussion on #932, we decided for now, for simplicity sake, we wish the filters and their dependencies to be included by default. 

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: DEV_GUIDE.md [hunk 1]
```diff
@@ -93,13 +93,13 @@ Build with the `dist` profile to create distribution artefacts (see [kroxyliciou
 The distribution includes the Kroxylicious-maintained Filter implementations (located under [kroxylicious-additional-filters](./kroxylicious-additional-filters)).
 
 ```shell
-mvn clean verify -Pdist -Dquick
+mvn clean package -Pdist -Dquick
```
Comment (k-wall) [inline DEV_GUIDE.md:96]: I wonder if the `dist` profile are has value.   As a maven user, least surprise is for `mvn package` to produce the tgz/zip artefact.  I don't really expect to need to turn something on.  I think the tgz/zip artefact should just occur by default.

Comment (robobario) [reply DEV_GUIDE.md:96]: sounds good to me

**Review by robobario — APPROVED** [review]
LGTM

File: DEV_GUIDE.md (hunk 1 above)
Comment (SamBarker) [reply DEV_GUIDE.md:96]: As a maven user I expect a build to produce only jars everything else just feels like extra weight that it doesn't need to do in the majority of build runs.

I.e. we only care about tgz when building the release itself. Hiding that behind a profile feels like an easy way to support that anyone who needs to build it locally. 

So I'd me -0 to just folding `-dist` in, but happy to go with consensus.

**Review by SamBarker — APPROVED** [review]
LGTM

File: Dockerfile [hunk 2]
```diff
@@ -9,7 +9,7 @@ FROM registry.access.redhat.com/ubi9/openjdk-17:1.15 AS builder
 USER root
 WORKDIR /opt/kroxylicious
 COPY . .
-RUN ./mvnw -B clean verify -Pdist,withAdditionalFilters -Dquick
+RUN ./mvnw -B clean package -Pdist -Dquick
```
Comment (SamBarker) [inline Dockerfile:12]: Ohh that should speed up the image build a bit, good find.

File: DEV_GUIDE.md (hunk 1 above)
Comment (k-wall) [reply DEV_GUIDE.md:96]: Thanks for the feedback, let's leave dist profile as it is.

**Pull request merged**